packages/wrangler/src/deployment-bundle/find-additional-modules.ts

```diff
@@ -98,6 +100,13 @@ export async function findAdditionalModules(
 			});
 		}
 	}
+
+	// The modules we find might also have sourcemaps associated with them, so when we go to copy
+	// them into the output directory we need to preserve the sourcemaps.
+	if (attachSourcemaps) {
```

Comment:
I don't particularly like this solution, but it is functional and minimally invasive. LMK if you think this isn't adequate and I can do some refactoring to make this less hacked-on.